apps/web/src/state/farmsV4/state/accountPositions/hooks/useAccountPositionDetailByPool.ts

```diff
@@ -0,0 +1,76 @@
+import { Protocol } from '@pancakeswap/farms'
+import { LegacyRouter } from '@pancakeswap/smart-router/legacy-router'
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { useCallback, useMemo } from 'react'
+import { Address, isAddressEqual } from 'viem'
+import { PoolInfo } from '../../type'
+import { getAccountV2LpDetails, getStablePairDetails } from '../fetcher'
+import { getAccountV3Positions } from '../fetcher/v3'
+import { PositionDetail, StableLPDetail, V2LPDetail } from '../type'
+
+type PoolPositionDetail = {
+  [Protocol.STABLE]: StableLPDetail
+  [Protocol.V2]: V2LPDetail
+  [Protocol.V3]: PositionDetail[]
+}
+
+export const useAccountPositionDetailByPool = <TProtocol extends keyof PoolPositionDetail>(
+  chainId: number,
+  account?: Address | null,
+  poolInfo?: PoolInfo,
+): UseQueryResult<PoolPositionDetail[TProtocol]> => {
+  const [currency0, currency1] = useMemo(() => {
+    if (!poolInfo) return [undefined, undefined]
+    const { token0, token1 } = poolInfo
+    return [token0.wrapped, token1.wrapped]
+  }, [poolInfo])
+  const protocol = useMemo(() => poolInfo?.protocol, [poolInfo])
+  const queryFn = useCallback(async () => {
+    if (protocol === 'v2') {
+      return getAccountV2LpDetails(
+        chainId,
+        account!,
+        currency0 && currency1 ? [[currency0.wrapped, currency1.wrapped]] : [],
+      )
+    }
+    if (protocol === 'stable') {
+      const stablePair = LegacyRouter.stableSwapPairsByChainId[chainId].find((pair) => {
+        return isAddressEqual(pair.stableSwapAddress, poolInfo?.lpAddress as Address)
+      })
+      return getStablePairDetails(chainId, account!, stablePair ? [stablePair] : [])
+    }
+    if (protocol === 'v3') {
+      return getAccountV3Positions(chainId, account!)
+    }
+    return Promise.resolve([])
+  }, [account, chainId, currency0, currency1, poolInfo, protocol])
```

Comment:
dep of poolInfo can change to poolInfo?.lpAddress to reduce changes, I figure out there are too much changes of object reference in the existing code